### PR TITLE
feat(cli): Updating function deployment status

### PR DESCRIPTION
**Pull request by byawitz**
## What does this PR do?

Show updates for function deployment in the Appwrite instance. 
In case of 30 POOL iterations, the status would change to continue to check on the console.

## Test Plan
Checked against an instance of Appwrite  with version `1.5.5` 👇

https://github.com/appwrite/sdk-generator/assets/316103/b72c43ca-83b1-4213-bb94-8066833fb70b

### CODE REVIEW

File: templates/cli/lib/commands/deploy.js.twig [hunk 1]
```diff
@@ -41,6 +41,7 @@ const {
 
 const STEP_SIZE = 100;      // Resources
 const POOL_DEBOUNCE = 2000; // Milliseconds
+const POOL_MAX_DEBOUNCE = 30; // Times
```
Comment (christyjacob4) [inline templates/cli/lib/commands/deploy.js.twig:44]: So I think this should be renamed to `POLL_` and not `POOL` lets also refactor the existing variable

File: templates/cli/lib/commands/deploy.js.twig [hunk 2]
```diff
@@ -450,12 +453,54 @@ const deployFunction = async ({ functionId, all, yes } = {}) => {
                     updaterRow.fail({ errorMessage: e.message ?? 'General error occurs please try again' })
             }
         }
+
+        if (deploymentCreated) {
+            try {
+                const deploymentId = response['$id'];
+                updaterRow.update({ status: 'Deploying', end: 'Checking deployment status...' })
+                let pool_checks = 0;
```
Comment (christyjacob4) [inline templates/cli/lib/commands/deploy.js.twig:461]: lets use camelcase to stay consistent

File: templates/cli/lib/commands/deploy.js.twig [hunk 3]
```diff
@@ -450,12 +453,54 @@ const deployFunction = async ({ functionId, all, yes } = {}) => {
...
+
+                while (true) {
+                    if (pool_checks >= POOL_MAX_DEBOUNCE) {
+                        updaterRow.update({ end: 'Deployment takes too long; check the console' })
```
Comment (christyjacob4) [inline templates/cli/lib/commands/deploy.js.twig:465]:
```suggestion
                        updaterRow.update({ end: 'Deployment is taking too long. Please check the console for more details.' })
```

File: templates/cli/lib/commands/deploy.js.twig [hunk 4]
```diff
@@ -450,12 +453,54 @@ const deployFunction = async ({ functionId, all, yes } = {}) => {
...
+
+
+                    const status = response['status'];
+                    if (status === 'ready') {
+                        updaterRow.update({ status: 'Deployed' });
+                        successfullyDeployed++;
+
+                        break;
+                    } else if (status === 'failed') {
+                        updaterRow.fail({ errorMessage: 'failed to deploy' });
```
Comment (christyjacob4) [inline templates/cli/lib/commands/deploy.js.twig:483]:
```suggestion
                        updaterRow.fail({ errorMessage: 'Failed to deploy.' });
```

File: templates/cli/lib/commands/deploy.js.twig [hunk 5]
```diff
@@ -450,12 +453,54 @@ const deployFunction = async ({ functionId, all, yes } = {}) => {
...
+
+                        break;
+                    } else {
+                        updaterRow.update({ status: 'Deploying', end: `current status: ${status}` })
```
Comment (christyjacob4) [inline templates/cli/lib/commands/deploy.js.twig:487]:
```suggestion
                        updaterRow.update({ status: 'Deploying', end: `Current status: ${status}` })
```

File: templates/cli/lib/commands/deploy.js.twig [hunk 6]
```diff
@@ -450,12 +453,54 @@ const deployFunction = async ({ functionId, all, yes } = {}) => {
...
+                    }
+
+                    pool_checks++;
+                    await new Promise(resolve => setTimeout(resolve, POOL_DEBOUNCE));
+                }
+            } catch (e) {
+                updaterRow.fail({ errorMessage: e.message ?? 'General error occurs please try again' })
```
Comment (christyjacob4) [inline templates/cli/lib/commands/deploy.js.twig:494]:
```suggestion
                updaterRow.fail({ errorMessage: e.message ?? 'Unknown error occurred. Please try again' })
```

File: templates/cli/lib/commands/deploy.js.twig [hunk 7]
```diff
@@ -450,12 +453,54 @@ const deployFunction = async ({ functionId, all, yes } = {}) => {
...
+            }
+        }
+
         updaterRow.stopSpinner();
     }));
 
     Updater.stop();
 
-    success(`Pushed ${successfullyDeployed} functions`);
+    success(`Pushed ${successfullyPushed} functions with ${successfullyDeployed} successfully deployed`);
```
Comment (christyjacob4) [inline templates/cli/lib/commands/deploy.js.twig:503]:
```suggestion
    success(`Successfully pushed ${successfullyPushed} functions.`);
```
I dont think the `successfullyDeployed` variable is required here?

Comment (byawitz) [reply templates/cli/lib/commands/deploy.js.twig:503]: The `successfullyDeployed` inform how many functions didn't fail in the Appwrite instance, could be changed to 
```js
success(`Successfully pushed & deployed ${successfullyDeployed} functions.`);
```

Comment (byawitz) [reply templates/cli/lib/commands/deploy.js.twig:503]: Or, we can use the refactor success message but only for deployed (pushed + deployed) functions
```js
success(`Successfully pushed ${successfullyDeployed} functions.`);
```

File: templates/cli/lib/commands/push.js.twig [hunk 8]
```diff
@@ -450,12 +453,54 @@ const pushFunction = async ({ functionId, all, yes } = {}) => {
...
+
+
+                    const status = response['status'];
+                    if (status === 'ready') {
+                        updaterRow.update({ status: 'Deployed' });
+                        successfullyDeployed++;
+
+                        break;
+                    } else if (status === 'failed') {
+                        updaterRow.fail({ errorMessage: 'Failed to deploy.' });
```
Comment (Meldiron) [inline templates/cli/lib/commands/push.js.twig:483]: Let's show logs, if possible. If it looks too weird, we can link to Console

Comment (byawitz) [reply templates/cli/lib/commands/push.js.twig:482]: image

File: templates/cli/lib/commands/push.js.twig [hunk 9]
```diff
@@ -449,12 +453,63 @@ const pushFunction = async ({ functionId, all, yes } = {}) => {
...
+
+    failedDeployments.forEach((failed) => {
+        const { name, deployment, $id } = failed;
+        const failUrl = `${globalConfig.getEndpoint().replace('/v1', '')}/console/project-${localConfig.getProject().projectId}/functions/function-${$id}/deployment-${deployment}`;
+
+        error(`Deployment of ${name} has failed. Check at ${failUrl} for more details\n`);
+    })
 
-    success(`Pushed ${successfullyPushed} functions`);
+    success(`Pushed ${successfullyPushed} functions with ${successfullyDeployed} successfully deployed`);
```
Comment (christyjacob4) [inline templates/cli/lib/commands/push.js.twig:512]:
```suggestion
    success(`Pushed ${successfullyPushed} functions with ${successfullyDeployed} successful deployments.`);
```

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**